src/deserializeSchema.ts

```diff
@@ -166,6 +207,27 @@ export type PairType = {
     ofRight: Type;
 };
 
+export function deserialFunction(source: Readable): ContractFunction {
+    const idx = deserialUint8(source);
+    switch (idx) {
+        case 0:
+            return {
+                parameter: deserialType(source),
+            };
+        case 1:
+            return {
+                returnValue: deserialType(source),
+            };
+        case 2:
+            return {
+                parameter: deserialType(source),
+                returnValue: deserialType(source),
+            };
+        default:
+            throw new Error('Incorrect index for function');
```

Comment:
```suggestion
            throw new Error('Incorrect tag for function');
```
We usually call them tags on the rust/haskell side. Is index used in JS?

Reply:
I renamed it to tag. Honestly didn't think about it and copied the var name from the rust implementation, where it was idx (I assume that's short for index :upside_down_face: )